File: microservices/company/src/main/java/com/illunex/invest/company/persistence/entity/Product.java
package com.illunex.invest.company.persistence.entity;

import lombok.*;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name= "product")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;
    String title;           // 제품명
    @Column(name = "description", length = 4000)
    String description;     // 제품 설명
    String viewMode;        // 공개 여부

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "productId")
    List<ProductImage> productImages = new ArrayList<>();   // 제품 이미지

    @OneToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "productId")
    List<Keyword> keywords = new ArrayList<>();

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    @JoinColumn(name = "companyId")
    Company company;
}

File: microservices/shop/src/main/kotlin/com/illunex/invest/shop/persistence/entity/Product.kt
package com.illunex.invest.shop.persistence.entity

import com.illunex.invest.shop.persistence.entity.enumable.ViewMode
import java.math.BigDecimal
import javax.persistence.*

@Entity
@Table(name = "product")
class Product {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    var title: String = ""
    var description: String = ""
    var imgUrl: String= ""
    var price: BigDecimal = BigDecimal.ZERO
    @Enumerated(value = EnumType.STRING)
    var viewMode: ViewMode = ViewMode.OPEN
    @ManyToMany(fetch = FetchType.LAZY)
    var plugins: List<Plugin> = ArrayList()

    constructor()

    constructor(id: Long?) {
        this.id = id
    }

    constructor(id: Long?, title: String, description: String, imgUrl: String, price: BigDecimal, viewMode: ViewMode, plugins: List<Plugin>) {
        this.id = id
        this.title = title
        this.description = description
        this.imgUrl = imgUrl
        this.price = price
        this.viewMode = viewMode
        this.plugins = plugins
    }
}